source/velocity_boundary_conditions/interface.cc

```diff
@@ -70,7 +70,8 @@ namespace aspect
     }
 
 
-
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
This is a virtual function implemented in a base class that acts as fallback calling the deprecated old function in case a user plugin does not implement this (new) interface. Because it calls a deprecated function the compiler generates a deprecation warning. 
The pragma was actually not my idea. I copied it from the heating model interface where we use the same approach for the same problem. So far I have seen that intel compilers omit a warning "Ignoring unknown pragma" or something like this, but it does not produce errors. I can try your suggestion instead Timo.